canal/canal.go

```diff
@@ -197,8 +230,52 @@ func (c *Canal) Ctx() context.Context {
 	return c.ctx
 }
 
+func (c *Canal) checkTableMatch(key string) bool {
+	// no filter, return true
+	if c.tableMatchCache == nil {
+		return true
+	}
+
+	c.tableLock.Lock()
```

Comment:
the lock scope is huge, just lock when you really need to operate the cache.

Btw, use read lock for reading cache and write lock for writing cache.

Reply:
done